### PR TITLE
serialize updates to healing tracker

### PR DESCRIPTION
**Pull request by krisis**
## Description
When healing is parallelized by setting _MINIO_HEAL_WORKERS environment
variable, multiple goroutines may race while updating the disk's healing
tracker. This change serializes only these concurrent updates using a
channel. Note, healing tracker is still not concurrency safe in other contexts.

## How to test this PR?
Test healing a fresh disk with `_MINIO_HEAL_WORKERS` set to a value > 1

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Optimization (provides speedup with no functional changes)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [ ] Fixes a regression (If yes, please add `commit-id` or `PR #` here)
- [ ] Documentation updated
- [ ] Unit tests added/updated

### CODE REVIEW

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-erasure.sh | :heavy_check_mark: |
| mint-dist-erasure.sh | :heavy_check_mark: |
| mint-compress-encrypt-dist-erasure.sh | :heavy_check_mark: |
| mint-pools.sh | :heavy_check_mark: |
| mint-fs.sh | :x: [more...](#a7386ab9-572e-49e6-99d0-ccc1a835cfd6) |
<a name="a7386ab9-572e-49e6-99d0-ccc1a835cfd6"></a>
### 15647-461102c/mint-fs.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-k8s1:31125
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0
RUN_ON_FAIL:          0

To get logs, run 'docker cp :/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... done in 1 seconds
(2/14) Running aws-sdk-java tests ... done in 1 seconds
(3/14) Running aws-sdk-php tests ... done in 43 seconds
(4/14) Running aws-sdk-ruby tests ... done in 4 seconds
(5/14) Running awscli tests ... done in 55 seconds
(6/14) Running healthcheck tests ... done in 0 seconds
(7/14) Running mc tests ... FAILED in 29 seconds
{
  "name": "mc",
  "duration": "11726",
  "function": "test_find",
  "status": "FAIL",
  "error": "mirror and list differs"
}
(7/14) Running minio-go tests ... done in 1 minutes and 15 seconds
(8/14) Running minio-java tests ... done in 19 seconds
(9/14) Running minio-js tests ... done in 58 seconds
(10/14) Running minio-py tests ... done in 47 seconds
(11/14) Running s3cmd tests ... done in 23 seconds
(12/14) Running s3select tests ... done in 2 seconds
(13/14) Running versioning tests ... done in 1 minutes and 1 seconds

Executed 13 out of 14 tests successfully.
```
Deleting image on docker hub
Deleting image locally

File: cmd/global-heal.go [hunk 1]
```diff
@@ -243,6 +243,53 @@ func (er *erasureObjects) healErasureSet(ctx context.Context, buckets []string,
...
+				if res.success {
+					tracker.ItemsHealed++
+					tracker.BytesDone += res.bytes
+				} else {
+					tracker.ItemsFailed++
+					tracker.BytesFailed += res.bytes
+				}
+
+			}
+		}()
```
Comment (harshavardhana) [inline cmd/global-heal.go]: when would be `close the entryCh` ? @krisis

File: cmd/global-heal.go [hunk 2]
```diff
@@ -277,10 +324,10 @@ func (er *erasureObjects) healErasureSet(ctx context.Context, buckets []string,
 					versionID: "",
 				}, madmin.HealItemObject)
 				if err != nil {
-					tracker.ItemsFailed++
+					entryCh <- healEntryFailure(0)
```
Comment (harshavardhana) [inline cmd/global-heal.go]: compete this with `context` @krisis  if context got canceled we don't have to block here infinitely.

**Review by harshavardhana — CHANGES_REQUESTED** [review]

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-erasure.sh | :heavy_check_mark: |
| mint-dist-erasure.sh | :heavy_check_mark: |
| mint-compress-encrypt-dist-erasure.sh | :heavy_check_mark: |
| mint-pools.sh | :heavy_check_mark: |
Deleting image on docker hub
Deleting image locally

**Review by harshavardhana — APPROVED** [review]

**Review by vadmeste — APPROVED** [review]
LGTM

**Review by klauspost — APPROVED** [review]

**Pull request merged**